Library/Homebrew/cmd/install.rb

```diff
@@ -135,13 +135,41 @@ def install
           raise "No devel block is defined for #{f.full_name}"
         end
 
-        current = f if f.installed?
-        current ||= f.old_installed_formulae.first
+        if f.keg_only? && !f.installed_prefixes.empty? && f.opt_prefix.symlink? && !ARGV.force?
+          # keg-only install is only possible when no ohter version is
+          # optlinked, because silent install can break dependencies. Therefore
+          # before performing other checks we need to be sure --force flag is
+          # passed.
+          opoo "#{f.full_name} is a keg-only and another version is optlinked."
+          puts "Use `brew install --force` if you want to install this version"
+        elsif (ARGV.build_head? && (version = f.latest_head_version) &&
+            !f.head_version_outdated?(version, fetch_head: ARGV.fetch_head?)) ||
+              (f.prefix.exist? && !f.prefix.children.empty?)
```

Comment:
Split some/all of these conditionals into variables. Similarly consider doing so for the `if` too.